apps/email/js/cards/cmp/autocomplete_source.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+define(function(require) {
+
+var contacts = navigator.mozContacts;
+
+function makeResult(query, contacts) {
+  return {
+    query: query,
+    contacts: contacts
+  };
+}
+
+/**
+ * Finds matches for the given query in the name and email for known people.
+ * Returns a promise that resolves to an object that has the query and a
+ * "match" array of mozContact entries. The match array will be a zero length
+ * array for no matches. See this bug about expanding the data source beyond
+ * contacts matches:
+ * https://bugzilla.mozilla.org/show_bug.cgi?id=1188710
+ * @param  {String} query the name/email query to use.
+ * @return {Promise}
+ */
+return function match(query) {
+  return new Promise(function(resolve, reject) {
+    if (!contacts || !query) {
+      return resolve(makeResult(query, []));
+    }
+
+    var request = contacts.find({
+      filterBy: ['name', 'email'],
```

Comment:
I went with adding all three of the name options: `filterBy: ['email', 'name', 'givenName', 'familyName']` -- that seems to allow matching on last name only, or if i do first name and a space matches on the whole name.